src/Submission/ReduxProjectSubmission.js

```diff
@@ -36,24 +38,58 @@ const submitToServer = (fullProject, methodIn = 'PUT') => (dispatch, getState) =
     // remove line break in json file
     file = file.replace(/\n/g, '');
   }
+
+
+  if (submission.file_type === 'text/tab-separated-values') {
+    const fileSplited = file.split(/\n/g);
+    if (fileSplited.length > lineLimit && lineLimit > 0) {
+      let fileHeader = fileSplited[0];
+      fileHeader += '\r';
+      let count = lineLimit;
+      let fileChunk = fileHeader;
+
+      for (let i = 1; i < fileSplited.length; i += 1) {
+        fileChunk += fileSplited[i];
+        fileChunk += '\r';
+        count -= 1;
+        if (count === 0) {
+          fileArray.push(fileChunk);
+          fileChunk = fileHeader;
```

Comment:
do you need to `+= '\r'` here?

Reply:
no `fileheader` already has `\r` at the end after line 47